### PR TITLE
prune application finalizer from global cluster

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Fixed: https://github.com/stolostron/backlog/issues/25315

### CODE REVIEW

**Comment by clyang82** [comment]
You need to run `make fmt` before submitting the PR. it is used to format the code to fix the code smell.

**Review by morvencao — COMMENTED** [review]
overall /LGTM
since you update role.yaml, you also need to run `make bundle` under operator directory to update the CSV.

**Comment by clyang82** [comment]
Do you have a following PR to handle the policy finalizer removal?

File: test/setup/hoh/hoh_setup.sh [hunk 1]
```diff
@@ -81,5 +81,27 @@ kubectl get cm mgh-images-config -n "$namespace" -oyaml
 kubectl get mgh multiclusterglobalhub -n "$namespace" -oyaml
 
 # wait for core components to be ready
-kubectl wait deployment -n open-cluster-management multicluster-global-hub-manager --for condition=Available=True --timeout=600s
-kubectl --context kind-hub1 wait deployment -n open-cluster-management multicluster-global-hub-agent --for condition=Available=True --timeout=600s
+SECOND=0
+while [[ -z $(kubectl get deploy -n $namespace multicluster-global-hub-manager --ignore-not-found) ]]; do
+  if [ $SECOND -gt 200 ]; then
```
Comment (clyang82) [inline test/setup/hoh/hoh_setup.sh:86]: Good catch ! Thanks.

**Comment by yanmxa** [comment]
> Do you have a following PR to handle the policy finalizer removal?

Yeah, I have another PR the process the policy.

**Pull request merged**